README.md

```diff
@@ -1,2 +1,135 @@
-# ros-pycharm-example
-A simple example for fully integrated Python development of a ROS / ROS2 project in PyCharm 
+# Developing Python ROS in PyCharm
+
+This is an example project to illustrate how use PyCharm as a fully integrated development environment for a ROS Python
+project by encapsulating the dependencies in a docker container.
+
+## The turtle example
+
+This example installs dependencies for the [`turtlesim`](http://wiki.ros.org/turtlesim) package on top of the base ROS
+image. It defines a new package called [`turtle_example`](./turtle_example) with a simple Python node
+called [`circler`](./turtle_example/scripts/circler)
+which publishes twist commands for a running turtlesim node to drive it in a circle.
+
+## Base ROS Docker images
+
+The majority of the background work is done in the
+repository [aica-technology/docker-images](https://github.com/aica-technology/docker-images). There, docker images are
+defined for specific versions of ROS and ROS2 that contain all of the necessary development dependencies, including an
+inbuilt SSH server. PyCharm will use the SSH server to access the remote environment.
+
+To begin, clone the [docker-images](https://github.com/aica-technology/docker-images) repository and follow the
+instructions to pull or build your base ROS workspace of choice.
+
+## PyCharm setup
+
+Prerequisites: Docker, PyCharm Professional 2021 or newer, base ROS docker image from AICA
+
+Step 1: Build the docker image with `docker build -t domire8/ros-pycharm-example .` (this will take a while the first
+time around).
+
+Step 2: Use the [`server.sh` script](https://github.com/aica-technology/docker-images/blob/master/scripts/server.sh) in
+the aica-technology/docker-images repository to start a background container for the appropriate user with a desired
+port number.
+
+In this example, we will use user `ros` for a noetic version image. We will also choose to use port 3456, though the
+specific number does not matter.
+
+```shell script
+path/to/aica-technology/docker-images/scripts/server.sh domire8/ros-pycharm-example --user ros --port 3456
+```
+
+Step 3: Open this directory in PyCharm or via command line
+
+```bash
+cd path/to/ros-pycharm-example && pycharm-professional .
+```
+
+Step 4: Go to `File` &rarr; `Settings` &rarr; `Project: ros-pycharm-example` &rarr; `Python Interpreter` and choose the
+option to add a new interpreter.
+
+![Add Interpreter](img/add_interpreter.png)
+
+Step 5: Choose to add a new `SSH Interpreter`, enter the server configuration as shown in the picture below, and click
+next. There might be a warning popping up if you want to continue connecting to the SSH server. Confirm with yes.
+
+![Configure Interpreter](img/configure_interpreter.png)
+
+Step 6: In the next window, change the interpreter path to `usr/bin/python3` and make sure that the option *
+Automatically upload project files to the server* is activated. Do not worry about the rest for now and click finish.
```

Comment:
>Do not worry about the rest for now and click finish.

You can actually set the path mapping here already, under "Sync folders:". This will link to the deployment mapping. But, you can still keep the next step as it sets the default connection.

Reply:
I agree, but I find it easier to do it under deployment, it's more explicit and clear there in my opinion